app/controllers/provider/passwords_controller.rb

```diff
@@ -48,10 +49,14 @@ def update
 
   private
 
+  def can_reset_password?
+    render_error 'Cannot reset password for Master Account. Please contact your system administrator.', status: 422 if domain_account.master?
```

Comment:
imo we should use cancancan for the permissions of this feature :smile:

Reply:
btw this is incompatible with this line:
https://github.com/3scale/porta/pull/1701/files#diff-2208b639adfb8b839f1a0ffb8ea3ef0aR60

Right now, it will NEVER reach that line... and we shouldn't have code that cannot get reached